File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        int vowels=0,maxvowels=0;
        for (int i = 0; i < k; i++) {
            if (isVowel(s.charAt(i))) {
                vowels++;
            }
        }

            maxvowels=vowels;
            for(int i=k;i<s.length();i++){
                if (isVowel(s.charAt(i-k))) {
                vowels--;
            }
                
                if (isVowel(s.charAt(i))) {
                vowels++;
            }
               
           maxvowels=Math.max(maxvowels,vowels);
           if(maxvowels==k){
            return k;
           }
            }     
            return maxvowels;
    }
            
    private boolean isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' ||
               c == 'o' || c == 'u';
    }
        
        
    

}

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        cnt=0
        maxcnt=0
        vowels="aeiou"
        for r in range(k):
            if(s[r] in vowels):
                cnt+=1
        maxcnt=cnt
        for r in range(k,len(s)):
            if(s[r] in vowels):
                cnt+=1
            if s[r-k] in vowels:
                cnt-=1
            maxcnt=max(maxcnt,cnt)
        return maxcnt
